src/engine/pieces/piece.ts

```diff
@@ -16,5 +29,65 @@ export default class Piece {
     public moveTo(board: Board, newSquare: Square) {
         const currentSquare = board.findPiece(this);
         board.movePiece(currentSquare, newSquare);
+        this.active = true;
+    }
+
+    public isValidMove(board: Board, nextSquare: Square) {
+
+        let pieceOnBoard = board.getPiece(nextSquare)
+        if (pieceOnBoard === undefined)
```

Comment:
```suggestion
        if (!pieceOnBoard)
```

can simplify this to this